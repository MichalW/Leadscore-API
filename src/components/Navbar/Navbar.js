import PropTypes from 'prop-types';
import React from 'react';
import {Menu} from 'semantic-ui-react';

const Navbar = ({signedIn, signOutRequest}) => (
  <Menu size="large">
    <Menu.Item name="home" onClick={() => {}} />

    {signedIn ? (
      <Menu.Item name="logout" onClick={signOutRequest} />
    ) : (
      <Menu.Item name="login" onClick={() => {}} />
    )}
  </Menu>
);

Navbar.propTypes = {
  signOutRequest: PropTypes.func.isRequired,
  signedIn: PropTypes.bool.isRequired,
};

export default Navbar;
