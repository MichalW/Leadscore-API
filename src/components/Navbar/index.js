import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Navbar from './Navbar';
import {signOutRequest} from '../../actions/authentication';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
});

const mapDispatchToProps = dispatch => ({
  signOutRequest: bindActionCreators(signOutRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
