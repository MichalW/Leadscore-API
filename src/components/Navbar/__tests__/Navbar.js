import React from 'react';
import renderer from 'react-test-renderer';

import Navbar from '../Navbar';

it('renders Navbar without crashing', () => {
  const tree = renderer
    .create(
      <Navbar
        signOutRequest={() => {}}
        signedIn
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
