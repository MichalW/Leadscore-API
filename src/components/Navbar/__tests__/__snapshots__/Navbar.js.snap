// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar without crashing 1`] = `
<div
  className="ui large menu"
>
  <a
    className="item"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    Logout
  </a>
</div>
`;
